catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -131,9 +136,17 @@ private void queryNext(int index) {
                 Map<String, Serializable> props = new HashMap<>();
                 // Avoid caching all results while dumping with native query mode
                 props.put("mode", "native");
-                response = catalog.query(filter.apply(index));
+                response = catalog.query(new QueryRequestImpl(new QueryImpl(filter,
+                        pageIndex,
+                        pageSize,
+                        SortBy.NATURAL_ORDER,
+                        false,
+                        5000l/*timeout? config? passed in?*/)));
```

Comment:
Constant for the default value and a setter to change it sounds like a good way to go here.